### PR TITLE
Fix Issue 2137 - Data not compressed on fly when adding to zip archive

### PR DESCRIPTION
**Pull request by RazvanN7**
This PR fixes the bug report by moving the code that does compression from the `build` method to the `addMember` method. As a result, `addMethod` is not `@safe` anymore due to the use of `compress` and `crc32`.

### CODE REVIEW

**Comment by CyberShadow** [comment]
> As a result, addMethod is not @safe anymore due to the use of compress and crc32.

Then wrap the unsafe zlib calls into a `@safe` wrapper. Don't break the API.

**Comment by RazvanN7** [comment]
@CyberShadow Done.

File: std/zip.d [hunk 1]
```diff
@@ -360,11 +360,37 @@ final class ZipArchive
     {
     }
 
-    /** Add de to the archive.
+    /** Add de to the archive. The file is compressed on fly.
```
Comment (CyberShadow) [inline std/zip.d]: I think "on the fly" is how the expression goes.

Comment (RazvanN7) [inline std/zip.d]: Fixed

**Pull request merged**